### PR TITLE
Agent testing environment ep 5922

### PR DESCRIPTION
**Pull request by maorlx**

### CODE REVIEW

File: .github/workflows/cluster_agent_tests.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        working-directory: ./pkg/cluster_agent
+        run: |
+          echo "::set-output name=matrixTestPath::`python ./cicd/generate_test_matrix.py`"
+
+  system_tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    needs: system_test_install
+    strategy:
+      fail-fast: false
```
Comment (ophiryael) [inline .github/workflows/cluster_agent_tests.yml:50]: Why is this ok for tests to fail?

File: .github/workflows/cluster_agent_tests.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.1.0
+        with:
+          cluster_name: test-cluster
+          config: ./pkg/cluster_agent/${{ steps.get-cluster-config-path.outputs.clusterConfig }}
+
+      - name: test-docker
+        run: which docker
+        working-directory: ./pkg/cluster_agent
```
Comment (ophiryael) [inline .github/workflows/cluster_agent_tests.yml]: Isn't it kind of redundant if we set up the kind cluster in the previous step?

File: .github/workflows/cluster_agent_tests.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+      - name: load-image-to-kind
+        run: kind load docker-image epsagon/cluster-agent:test --name test-cluster
+
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - name: Install dependencies # no if here since we still need to install from pip cache
```
Comment (ophiryael) [inline .github/workflows/cluster_agent_tests.yml]: Didn't understand the comment.

File: pkg/cluster_agent/tests/system/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<p align="center">
+  <a href="https://epsagon.com" target="_blank" align="center">
+    <img src="https://cdn2.hubspot.net/hubfs/4636301/Positive%20RGB_Logo%20Horizontal%20-01.svg" width="300">
+  </a>
+  <br />
+</p>
+
+# Epsagon Kubernetes Cluster Agent - System tests
+
+To run the ckuster agent system tests, you need the kubectl context to be set to a
```
Comment (ophiryael) [inline pkg/cluster_agent/tests/system/README.md]: ckuster => cluster

File: pkg/cluster_agent/tests/test_agent_deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+apiVersion: v1
```
Comment (ophiryael) [inline pkg/cluster_agent/tests/test_agent_deployment.yaml]: Where is this deployment used?

**Review by ophiryael — COMMENTED** [review]
Great testing infrastructure, supporting different kind cluster config for each test was a good idea.
Left some small comments.

File: pkg/cluster_agent/tests/test_agent_deployment.yaml (hunk 5 above)
Comment (maorlx) [reply pkg/cluster_agent/tests/test_agent_deployment.yaml]: not used, deleting redundant file. thanks

File: .github/workflows/cluster_agent_tests.yml (hunk 2 above)
Comment (maorlx) [reply .github/workflows/cluster_agent_tests.yml]: definitely, used as debugging for me. Removing

File: .github/workflows/cluster_agent_tests.yml (hunk 1 above)
Comment (maorlx) [reply .github/workflows/cluster_agent_tests.yml:50]: ah its not, it just means that GitHub will run all the tests and not fail immediately if one of the tests fail (so you can see all the test which failed)

**Review by ophiryael — APPROVED** [review]

**Pull request merged**